File: unit-test-practice-4/src/test/java/dojo/CalculatorTest.java
package dojo;

import static org.assertj.core.api.Assertions.assertThat;
import static org.assertj.core.api.Assertions.catchThrowable;

import org.testng.annotations.Test;

public class CalculatorTest {

	private Calculator calculator = new Calculator();
	
	@Test
	public void should_throw_exception_for_null_input() throws Exception {
		Throwable ex = catchThrowable(() -> calculator.calculate(null));
		assertThat(ex).isInstanceOf(NullPointerException.class);
	}
	
	@Test(enabled=false)
	public void should_throw_exception_for_empty_string() throws Exception {
		Throwable ex = catchThrowable(() -> calculator.calculate(""));
		assertThat(ex).isInstanceOf(IllegalArgumentException.class);
	}
	
	@Test(enabled=false)
	public void should_throw_exception_for_input_with_only_spaces() throws Exception {
		Throwable ex = catchThrowable(() -> calculator.calculate(" \t\n"));
		assertThat(ex).isInstanceOf(IllegalArgumentException.class);
	}
	
	@Test(enabled=false)
	public void should_throw_exception_for_non_numeric_input() throws Exception {
		Throwable ex = catchThrowable(() -> calculator.calculate("a"));
		assertThat(ex).isInstanceOf(IllegalArgumentException.class);
	}

	@Test(enabled=false)
	public void should_return_the_only_integer_input() throws Exception {
		int result = calculator.calculate("1");
		assertThat(result).isEqualTo(1);
	}
	
	@Test(enabled=false)
	public void should_add_two_integers() throws Exception {
		int result = calculator.calculate("1+2");
		assertThat(result).isEqualTo(3);
	}
	
	@Test(enabled=false)
	public void should_add_multiple_integers() throws Exception {
		int result = calculator.calculate("1+2+3+4+5+6");
		assertThat(result).isEqualTo(21);
	}
	
	@Test(enabled=false)
	public void should_skip_spaces_between_numbers() throws Exception {
		int result = calculator.calculate("1 + 2\n+ 3\t+ 4");
		assertThat(result).isEqualTo(10);
	}
	
	@Test(enabled=false)
	public void should_do_subtraction() throws Exception {
		int result = calculator.calculate("1 - 2");
		assertThat(result).isEqualTo(-1);
	}

	@Test(enabled=false)
	public void should_handle_negative_number() throws Exception {
		int result = calculator.calculate("1 + -2");
		assertThat(result).isEqualTo(-1);
	}
}

File: unit-test-practice-4/src/test/csharp/dojo.Tests/CalculatorTest.cs
namespace dojo.Tests;

public class Tests
{
    [TestFixture]
    public class CalculatorTest
    {
        private Calculator calculator = new Calculator();

        [Test]
        public void Should_throw_exception_for_null_input()
        {
            Assert.Throws<ArgumentNullException>(() => calculator.Calculate(null));
        }

        [Test]
        public void Should_throw_exception_for_empty_string()
        {
            Assert.Throws<ArgumentException>(() => calculator.Calculate(""));
        }

        [Test]
        public void Should_throw_exception_for_input_with_only_spaces()
        {
            Assert.Throws<ArgumentException>(() => calculator.Calculate(" \t\n"));
        }

        [Test]
        public void Should_throw_exception_for_non_numeric_input()
        {
            Assert.Throws<FormatException>(() => calculator.Calculate("a"));
        }

        [Test]
        public void Should_return_the_only_integer_input()
        {
            int result = calculator.Calculate("1");
            Assert.AreEqual(1, result);
        }

        [Test]
        public void Should_add_two_integers()
        {
            int result = calculator.Calculate("1+2");
            Assert.AreEqual(3, result);
        }

        [Test]
        public void Should_add_three_integers()
        {
            int result = calculator.Calculate("1+2+3");
            Assert.AreEqual(6, result);
        }

        [Test]
        public void Should_add_multiple_integers()
        {
            int result = calculator.Calculate("1+2+3+4+5+6");
            Assert.AreEqual(21, result);
        }

        [Test]
        public void Should_skip_spaces_between_numbers()
        {
            int result = calculator.Calculate("1 + 2\n+ 3\t+ 4");
            Assert.AreEqual(10, result);
        }

        [Test]
        public void Should_handle_leading_plus_sign()
        {
            int result = calculator.Calculate("+1 + 2");
            Assert.AreEqual(3, result);
        }

        [Test]
        public void Should_handle_leading_minus_sign()
        {
            int result = calculator.Calculate("-1 + 2");
            Assert.AreEqual(1, result);
        }

        [Test]
        public void Should_handle_leading_plus_and_minus_sign()
        {
            int result = calculator.Calculate("-1 + +2");
            Assert.AreEqual(1, result);
        }

        [Test]
        public void Should_do_subtraction()
        {
            int result = calculator.Calculate("1 - 2");
            Assert.AreEqual(-1, result);
        }

        [Test]
        public void Should_handle_negative_number()
        {
            int result = calculator.Calculate("1 + -2-1");
            Assert.AreEqual(-2, result);
        }
    }
}
